### PR TITLE
support multiple target frameworks

### PR DESCRIPTION
**Pull request by patricklee2**

### CODE REVIEW

File: lib/templates/deploy.batch.console.netcore.template [hunk 1]
```diff
@@ -10,7 +10,7 @@ call :ExecuteCmd dotnet restore "%DEPLOYMENT_SOURCE%\{RestoreArguments}"
 IF !ERRORLEVEL! NEQ 0 goto error
 
 :: 2. Build and publish
-call :ExecuteCmd dotnet publish "%DEPLOYMENT_SOURCE%\{DotnetpublishArguments}" --output "%DEPLOYMENT_TEMP%\app_data\jobs\continuous\deployedJob" --configuration Release
+call :ExecuteCmd dotnet publish "%DEPLOYMENT_SOURCE%\{DotnetpublishArguments}" --output "%DEPLOYMENT_TEMP%\app_data\jobs\continuous\deployedJob" --configuration Release -f {TargetFramework}
```
Comment (suwatch) [inline lib/templates/deploy.batch.console.netcore.template]: So we always will pass target framework?   this is change in behavior correct?

**Comment by patricklee2** [comment]
Yes, passing in the framework is new

File: lib/generator._js [hunk 2]
```diff
@@ -119,6 +119,15 @@ function ScriptGenerator(repositoryRoot, projectType, projectPath, solutionPath,
   this.noDotDeployment = noDotDeployment;
   this.noSolution = noSolution;
   this.absoluteSitePath = path.join(this.repositoryRoot, this.sitePath);
+  if (!targetFramework && targetFramework != null)
+  {
```
Comment (suwatch) [inline lib/generator._js]: one more check for empty string

Comment (suwatch) [reply lib/generator._js]: 2.1 and 2.2 don't pass targetframework arg.

Comment (suwatch) [reply lib/generator._js]: double check deploy.cmd before and after your change

File: lib/generator._js [hunk 3]
```diff
@@ -119,6 +119,15 @@ function ScriptGenerator(repositoryRoot, projectType, projectPath, solutionPath,
...
+    this.targetFramework = " -f " + targetFramework;
+  }
```
Comment (suwatch) [inline lib/generator._js]: taking our prefix space

**Pull request merged**